src/visualizers/FactorFence.ts

```diff
@@ -0,0 +1,738 @@
+                    }
+                    bottomColor = gradient.bottom
+                    topColor = gradient.top
+
+                    // determine where to put lower left corner
+                    const barStart = this.graphCorner.copy()
+
+                    // move over based on which term
+                    const moveOver = this.recSpace.copy()
+                    moveOver.mult(myIndex - this.first)
+
+                    // move up based on cumulative height so far
+                    const moveUp = this.sketch.createVector(0, -cumulHt)
+                    barStart.add(moveOver)
+                    barStart.add(moveUp)
+
+                    // figure out upper right corner
+                    const barDiag = this.sketch.createVector(
+                        this.recWidth,
+                        recHeight
+                    )
+
+                    // draw the rectangle
+                    this.grad_rect(
+                        barStart.x,
+                        barStart.y,
+                        barDiag.x,
+                        barDiag.y,
+                        topColor,
+                        bottomColor
+                    )
+
+                    // if the mouse is over the rectangle being drawn
+                    // then we make note of the prime factor
+                    // and term we are hovering over
+                    const testVec = this.sketch.createVector(
+                        this.sketch.mouseX,
+                        this.sketch.mouseY
+                    )
+                    testVec.mult(1 / this.scaleFactor).sub(barStart)
+                    testVec.y = testVec.y * mySign
+                    const barDiagAbs = barDiag.copy()
+                    barDiagAbs.y = -barDiagAbs.y * mySign
+                    if (
+                        testVec.x >= 0
+                        && testVec.x <= barDiagAbs.x
+                        && testVec.y <= 0
+                        && testVec.y >= barDiagAbs.y
+                    ) {
+                        this.mousePrime = factor.prime
+                        this.mouseIndex = myIndex
+                        this.mouseOn = true
+                    }
+
+                    // move upward in preparation for next bar
+                    cumulHt += recHeight
+                }
+            }
+        }
+    }
+
+    resetLoop() {
+        this.frame = 0
+        this.sketch.loop()
+    }
+
+    keyPressed() {
+        this.resetLoop()
+    }
+
+    mouseMoved() {
+        this.resetLoop()
+    }
+
+    mouseWheel(event: MouseEvent) {
+        let scaleFac = 1
+        if (event instanceof WheelEvent) {
+            if (event.deltaY > 0) scaleFac = 1.03
+            else scaleFac = 0.97
+        }
+        this.scaleFactor *= scaleFac
+        this.graphCorner.y = this.graphCorner.y / scaleFac
+        this.resetLoop()
+    }
+
+    // set highlight prime by click
+    mouseClicked() {
+        if (this.mouseOn) {
```

Comment:
Seems to have the same issue that it fires when I click on the parameter tabs.

Reply:
yes, we have a separate comment chain about this now, so could resolve this

Reply:
Let's wait, to make sure the eventual fix for the wheel also fixes this issue.

Reply:
Seems to be solved by the general event filters. Resolving.